.github/workflows/sync-with-upstream.yaml

```diff
@@ -0,0 +1,76 @@
+name: 'Sync with Upstream'
+
+on:
+  schedule:
+   - cron:  '0 6 * * *'
+    #scheduled at 06:00 everyday
+  workflow_dispatch: {}
+
+permissions:
+  actions: write
+  contents: write
+
+env:
+  GH_USER: "github-actions[bot]"
+  GH_EMAIL: "<41898282+github-actions[bot]@users.noreply.github.com>"
+
+jobs:
+  sync-with-upstream-main:
+    runs-on: ubuntu-latest
+    name: Sync with upstream main
+    steps:
+    - name: Generate GitHub app token
+      id: github-app-token
+      uses: actions/create-github-app-token@5d869da34e18e7287c1daad50e0b8ea0f506ce69 # v1.11.0
+      with:
+        app-id: ${{ secrets.APP_ID }}
+        private-key: ${{ secrets.APP_PRIVATE_KEY }}
+    - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
+      with:
+        fetch-depth: 0
+        token: ${{ steps.github-app-token.outputs.token }}
+    - name: Generate GitHub app token
+      id: github-app-token
+      uses: actions/create-github-app-token@5d869da34e18e7287c1daad50e0b8ea0f506ce69 # v1.11.0
+      with:
+        app-id: ${{ secrets.APP_ID }}
+        private-key: ${{ secrets.APP_PRIVATE_KEY }}
+    - name: Sync upstream changes
+      id: sync
+      uses: aormsby/Fork-Sync-With-Upstream-action@v3.4.1
```

Comment:
[isn't this just running](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork#syncing-a-fork-branch-with-the-github-cli):

```
gh repo sync owner/cli-fork -b BRANCH-NAME
```

If it's just that maybe makes sense not to pull in another dependency?

Reply:
I don't have a strong opinion on this. This action seems simple and does everything we need and don't need to use `gh` cli command. I would keep it and change only if we have some issues